src/main/java/org/quiltmc/loader/impl/QuiltLoaderImpl.java

```diff
@@ -117,7 +118,14 @@ public void setGameProvider(GameProvider provider) {
 
 	private void setGameDir(Path gameDir) {
 		this.gameDir = gameDir;
-		this.configDir = gameDir.resolve("config");
+		String configSubDir = System.getProperty("quilt.configSubDir");
+		this.configDir = gameDir.resolve(configSubDir == null ? "config" : configSubDir);
```

Comment:
```suggestion
		this.configDir = gameDir.resolve(configSubDir == null || configSubDir.isEmtpy() ? "config" : configSubDir);
```
This may be empty if left as `-Dquilt.configSubDir` with no value